drivers/storage/portworx/components_test.go

```diff
@@ -11120,6 +11120,41 @@ func TestTelemetryEnable(t *testing.T) {
 	require.Len(t, telemetryConfigMap.OwnerReferences, 1)
 	require.Equal(t, cluster.Name, telemetryConfigMap.OwnerReferences[0].Name)
 	require.Equal(t, expectedConfigMap.Data, telemetryConfigMap.Data)
+
+	// Proxy config map
+	configMap := &v1.ConfigMap{}
+	err = testutil.Get(k8sClient, configMap, component.CollectorProxyConfigMapName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Len(t, configMap.OwnerReferences, 1)
+	require.Equal(t, cluster.Name, configMap.OwnerReferences[0].Name)
+
+	// Collector config map
+	configMap = &v1.ConfigMap{}
+	err = testutil.Get(k8sClient, configMap, component.CollectorConfigMapName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Len(t, configMap.OwnerReferences, 1)
+	require.Equal(t, cluster.Name, configMap.OwnerReferences[0].Name)
+
+	// Collector role
+	role := &rbacv1.Role{}
+	err = testutil.Get(k8sClient, role, component.CollectorRoleName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Len(t, role.OwnerReferences, 1)
+	require.Equal(t, cluster.Name, role.OwnerReferences[0].Name)
+
+	// Collector role binding
+	roleBinding := &rbacv1.RoleBinding{}
+	err = testutil.Get(k8sClient, roleBinding, component.CollectorRoleBindingName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Len(t, roleBinding.OwnerReferences, 1)
+	require.Equal(t, cluster.Name, roleBinding.OwnerReferences[0].Name)
+
+	// Collector deployment
+	deployment := &appsv1.Deployment{}
+	err = testutil.Get(k8sClient, deployment, component.CollectorDeploymentName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Len(t, deployment.OwnerReferences, 1)
+	require.Equal(t, cluster.Name, deployment.OwnerReferences[0].Name)
```

Comment:
Can we add unit tests for disabling metrics collector?
Like we did for repo change, can we add more tests for metrics collector too. For reference - https://github.com/libopenstorage/operator/pull/358#discussion_r683685356

Reply:
That piece of change is a little hard to maintain, i am thinking about using deepEqual to diff the two deployments spec and decide whether to update it. i just posted it in another commit, what do you think about this approach, in this case we don't need to worry what element is changed, any change in the spec would trigger update.

Reply:
DeepEqual would be ideal, but unfortunately it does not work. When you save a deployment object in k8s it adds a lot of defaults, which causes the DeepEqual to fail. That's why we resorted to this solution. Even I don't like it but comparing complicated objects like deployment, daemonset is quite tricky. You can try it on a cluster and check.

Anyway, the above comment was to add more tests for the deployment changes and checking disable.

Reply:
I think Kubernetes only adds to other part of the object (such as status) but does not touch the spec, let me test it and see. 

if we do diff of the entire spec then we dont have to worry about each field of change, which is essentially equal change. I would add disabling test first.

Reply:
@piyush-nimbalkar this is nice discussion, thanks for the tips. I finally found there is existing API in k8s that can do deep semantics deep equal, which is exactly what we need. 

in 	"k8s.io/apimachinery/pkg/api/equality"

// DeepDerivative is similar to DeepEqual except that unset fields in a1 are
// ignored (not compared). This allows us to focus on the fields that matter to
// the semantic comparison.
//
// The unset fields include a nil pointer and an empty string.
func (e Equalities) DeepDerivative(a1, a2 interface{}) bool {